### DISCUSSION TITLE
removing localStorage

### DISCUSSION THREAD

**Discussion by HilmanKhairr** [SafeExamBrowser/seb-mac#398] [General]
hii.. I want to ask, how to delete localstorage on seb mac? thanks

**Comment by danschlet** [comment]
There is no SEB setting for that. Normally I guess web applications delete the items they created in localstorage themselves. Why do you need such a feature in SEB?

**Reply by HilmanKhairr** [reply]
Sorry, I don't need the feature but just want to know how to do it manually. I'm not a mac user and I can't find how to remove it

**Reply by t00** [reply]
> There is no SEB setting for that. Normally I guess web applications delete the items they created in localstorage themselves. Why do you need such a feature in SEB?

In non-BYOD environments clearing all traces of a previous student makes perfect sense and when done on SEB will guarantee clean slate for a next student to sign in. Currently it works fine on Windows using removeBrowserProfile option but on a Mac and iOS it does not (removeLocalStorage option is no longer used?).

Also, when using quit URL feature, the browser is closed on Windows and session removed when using removeBrowserProfile. On iOS, quit URL closes the exam session but keeps the user signed in and does not in fact quit SEB - just restarts it.

Quit URL option itself might benefit from allowing to wipe localStorage/cookies/localDB etc when student finishes the exam. When Quit feature is fixed on iOS (closes the app) it will be on par with MacOS so adding an option to clear browser data on quit would make sense.

Having said that, seb configuration with an option to clear browser data on start should work in most cases, except storage integrity can be theoretically compromised.

**Reply by danschlet** [reply]
>On iOS, quit URL closes the exam session but keeps the user signed in and does not in fact quit SEB - just restarts it.

No, session cookies are deleted. As long as you don't change SEB's default settings (SEB in-app Settings / Exam Session / Session Cookies / Clear on Session End), users should not stay logged in. If that's the case it would be a bug, then provide me with a reproducible example please.

On iOS, apps themselves never quit or terminate itself, that would be against Apple's Human Interface Guidelines, which I follow religiously. That's why SEB quits the session and starts a fresh one (using client settings if those were configured, for example to show a portal page of an institution). iOS itself removes apps from memory when needed.
